### PR TITLE
Download configcli from its own git repo.

### PR DESCRIPTION
**Pull request by kcsraju**
I tried git submodules but that complicated things really fast. So, I think the easiest way to deal with configcli in kubedirector is to download the supported version during the build process. ConfigCli developers can always replace `tmp/_output/configcli.tgz` to test new changes. Please note that I am pointing to ConfigCli-v0.5 for now but will come back to update it soon.

### Testing ###
Spark cluster deployment succeeded.

```
time="2019-02-25T18:13:21Z" level=info msg="cluster{default/spark-instance}: initial config ongoing for member{kd-8cr8r-0} in role{worker}"
time="2019-02-25T18:13:21Z" level=info msg="cluster{default/spark-instance}: initial config ongoing for member{kd-8cr8r-1} in role{worker}"
time="2019-02-25T18:13:21Z" level=info msg="cluster{default/spark-instance}: reading file{/opt/guestconfig/configure.status} in pod{kd-2zmp8-0}"
time="2019-02-25T18:13:21Z" level=info msg="cluster{default/spark-instance}: creating file{/etc/guestconfig/configmeta.json} in pod{kd-2zmp8-0}"
time="2019-02-25T18:13:24Z" level=info msg="cluster{default/spark-instance}: creating file{/tmp/configcli.tgz} in pod{kd-2zmp8-0}"
time="2019-02-25T18:13:26Z" level=info msg="cluster{default/spark-instance}: running configcli setup in pod{kd-2zmp8-0}"
time="2019-02-25T18:13:28Z" level=info msg="cluster{default/spark-instance}: running app config setup in pod{kd-2zmp8-0}"
time="2019-02-25T18:13:30Z" level=info msg="cluster{default/spark-instance}: running app config in pod{kd-2zmp8-0}"
time="2019-02-25T18:13:32Z" level=info msg="cluster{default/spark-instance}: initial config ongoing for member{kd-2zmp8-0} in role{jupyter}"
time="2019-02-25T18:13:32Z" level=info msg="cluster{default/spark-instance}: reading file{/opt/guestconfig/configure.status} in pod{kd-hcvfc-0}"
time="2019-02-25T18:13:32Z" level=info msg="cluster{default/spark-instance}: initial config done for member{kd-hcvfc-0} in role{controller}"
time="2019-02-25T18:13:32Z" level=info msg="cluster{default/spark-instance}: reading file{/opt/guestconfig/configure.status} in pod{kd-8cr8r-1}"
time="2019-02-25T18:13:32Z" level=info msg="cluster{default/spark-instance}: reading file{/opt/guestconfig/configure.status} in pod{kd-8cr8r-0}"
time="2019-02-25T18:13:32Z" level=info msg="cluster{default/spark-instance}: initial config done for member{kd-8cr8r-0} in role{worker}"
time="2019-02-25T18:13:32Z" level=info msg="cluster{default/spark-instance}: initial config done for member{kd-8cr8r-1} in role{worker}"
time="2019-02-25T18:13:32Z" level=info msg="cluster{default/spark-instance}: reading file{/opt/guestconfig/configure.status} in pod{kd-2zmp8-0}"
time="2019-02-25T18:13:33Z" level=info msg="cluster{default/spark-instance}: initial config done for member{kd-2zmp8-0} in role{jupyter}"
time="2019-02-25T18:13:33Z" level=info msg="cluster{default/spark-instance}: stable"
```

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ That's far from the only user-visible change though! The entire list of improvem
 
 * The "deploy/example_catalog" directory has gained a new very simple example app (vanilla CentOS 7) and a new complicated one (CDH 5.14.2 with Pig, Hive, and Hue support).
 
-* The "configcli" tool used in app setup is now included in this repo, in the "nodeprep" directory.
+* The "configcli" tool used in app setup is a separate git repository at https://github.com/bluek8s/configcli
```
Comment (joel-bluedata) [inline HISTORY.md]: This is the history of particular released versions, so no revisionism. :-)

Comment (joel-bluedata) [reply HISTORY.md]: If you want, you can start a new v0.3.0 section at the top of the file (with a ??? release date) that incudes moving configcli as a bullet point feature.

Rebase to the current state of master branch first tho.

File: Makefile [hunk 2]
```diff
@@ -21,11 +23,14 @@ sedseparator = ''
 endif
 
 build_dir = 'tmp/_output'
+configcli_dest := $(build_dir)/configcli.tgz
 
-build: pkg/apis/kubedirector.bluedata.io/v1alpha1/zz_generated.deepcopy.go | $(build_dir)
-	@echo
-	@echo \* Creating node prep package...
-	tar cfzP tmp/_output/nodeprep.tgz nodeprep
+configcli:  | $(build_dir)
```
Comment (joel-bluedata) [inline Makefile]: "build" should be the first/default target in the Makefile.

File: Makefile [hunk 3]
```diff
@@ -21,11 +23,14 @@ sedseparator = ''
...
+	@if [ -e $(configcli_dest) ]; then exit 0; fi;                             \
+echo "\* Downloading configcli package ...";                                   \
```
Comment (joel-bluedata) [inline Makefile]: We can go ahead and indent these script lines out (with spaces) to pretty things up somewhat, as with other similar stuff below.

File: pkg/reconciler/types.go [hunk 4]
```diff
@@ -70,11 +70,13 @@ const (
...
 const (
 	configMetaFile     = "/etc/guestconfig/configmeta.json"
-	nodePrepSrcFile    = "/root/nodeprep.tgz"
-	nodePrepDestFile   = "/tmp/nodeprep.tgz"
-	nodePrepInstallCmd = `cd /tmp;tar xzf nodeprep.tgz;
-	chmod +x /tmp/nodeprep/install;/tmp/nodeprep/install;
-	rm -rf /tmp/nodeprep;rm -f /tmp/nodeprep.tgz`
+	nodePrepSrcFile    = "/root/configcli.tgz"
+	nodePrepDestFile   = "/tmp/configcli.tgz"
+	nodePrepInstallCmd = `cd /tmp;tar xzf configcli.tgz;
```
Comment (swamibluedata) [inline pkg/reconciler/types.go]: why not change these const names too

File: pkg/reconciler/members.go [hunk 5]
```diff
@@ -703,7 +703,7 @@ func appConfig(
 	if configmetaErr != nil {
 		return true, configmetaErr
 	}
-	// Set up nodeprep package for this member (if not set up already).
+	// Set up configcli package for this member (if not set up already).
 	prepErr := setupNodePrep(cr, podName)
```
Comment (swamibluedata) [inline pkg/reconciler/members.go]: just few more that refers to nodeprep

Comment (kcsraju) [reply pkg/reconciler/members.go]: good catch!

**Pull request merged**